### PR TITLE
[AppConfig] Added new features for Snapshot Configurations

### PR DESCRIPTION
**Pull request by mssfang**
# Description

Added snapshot feature to the AppConfiguration SDK. Such as, To Create, Get, Update(Archive and Recover), List snapshots.
Added APIs to get all configuration settings for a snapshot.

TODO: Update README 

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/models/CompositionType.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Collection;
+
+/**
+ * The composition type describes how the key-values within the snapshot are composed. The 'all' composition type
+ * includes all key-values. The 'group_by_key' composition type ensures there are no two key-values containing the same
+ * key.
+ */
+public final class CompositionType extends ExpandableStringEnum<CompositionType> {
```
Comment (mssfang) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/models/CompositionType.java:17]: use autotest customization

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/models/SnapshotSettingFilter.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+package com.azure.data.appconfiguration.models;
+
+import com.azure.core.annotation.Fluent;
+
+/**
+ * SnapshotFilter
+ */
+@Fluent
+public final class SnapshotSettingFilter {
```
Comment (mssfang) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/models/SnapshotSettingFilter.java:12]: use autorest renaming customizaton

File: sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,11 @@
 
 ### Features Added
 
+- Added new feature, `Configuration Setting Snapshot` to the library. You can create, get,
+  update(archive and recover) a snapshot, and list snapshots.
+- Added new APis to support listing configuration settings by given snapshot name and setting
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md:9]:
```suggestion
- Added new APIs to support listing configuration settings by given snapshot name and setting 
```

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/CreateSnapshotUtilClient.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+    /**
+     * Get the non-null {@link Context}. The default value is {@link Context#NONE}.
+     *
+     * @param context It offers a means of passing arbitrary data (key-value pairs) to pipeline policies.
+     * Most applications do not need to pass arbitrary data to the pipeline and can pass Context.NONE or null.
+     *
+     * @return The Context.
+     */
+    private Context getNotNullContext(Context context) {
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/CreateSnapshotUtilClient.java:205]: Should this be turned into a static utility

**Review by alzimmermsft — APPROVED** [review]

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/CreateSnapshotUtilClient.java (hunk 4 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/CreateSnapshotUtilClient.java:205]: Should it be? Each instance should have an independent context I think.

**Pull request merged**